In [1]:
import pandas
import numpy
import matplotlib.pyplot as plt

data = pandas.read_csv('StudentStatistics.csv', sep =';')
data.head()

RespNr   Location   OAA_ObjCourse                OAA_ObjClass  \
0     1.0  Rotterdam  Fully Disagree              Fully Disagree   
1     2.0    Haarlem        Disagree                    Disagree   
2     3.0     Diemen     Fully agree                 Fully agree   
3     4.0  Rotterdam  Fully Disagree  Neither disagree nor agree   
4     5.0    Haarlem        Disagree                       Agree   

   OAA_CourseExec               OAA_RelActObj               OAA_RelActExa  \
0  Fully Disagree                    Disagree              Fully Disagree   
1             NaN              Fully Disagree  Neither disagree nor agree   
2           Agree                 Fully agree                 Fully agree   
3        Disagree  Neither disagree nor agree  Neither disagree nor agree   
4  Fully Disagree  Neither disagree nor agree                 Fully agree   

    OAA_RelObjExa             OAA_LearProcAct            OAA_LearProcPrep  \
0  Fully Disagree              Fully Disagree              Fully Disagree   
1           Agree                    Disagree  Neither disagree nor agree   
2     Fully agree                 Fully agree                       Agree   
3  Fully Disagree              Fully Disagree  Neither disagree nor agree   
4     Fully agree  Neither disagree nor agree                 Fully agree   

   ... Mix_IntDiscus  Mix_IntPrep Mix_IntExcel Over_Grade Gen_Gender Gen_Age  \
0  ...    Not at all  not so much  not so much       20.0     Female    22.0   
1  ...   not so much     somewhat     somewhat       50.0       Male     NaN   
2  ...     Very much     somewhat     somewhat       80.0       Male    37.0   
3  ...           NaN      neutral      neutral       15.0     Female    24.0   
4  ...    Not at all     somewhat    Very much       40.0       Male    19.0   

     Gen_SecSchool Gen_Classes Gen_NumberSubj Gen_Time  
0              NaN         NaN    Fully agree  20 < 30  
1  The Netherlands         6.0       Disagree  10 < 20  
2           Africa         7.0          Agree  10 < 20  
3  The Netherlands         6.0          Agree  10 < 20  
4  The Netherlands         7.0    Fully agree  10 < 20  

[5 rows x 50 columns]

In [2]:
selected_data = data[['Teach_Motivate', 'Teach_LinkTheory', 'Teach_StimAsk', 'Teach_Avail']]
selected_data

Teach_Motivate            Teach_LinkTheory  \
0               Fully Disagree              Fully Disagree   
1                     Disagree                    Disagree   
2                  Fully agree                 Fully agree   
3               Fully Disagree  Neither disagree nor agree   
4               Fully Disagree              Fully Disagree   
5                  Fully agree                 Fully agree   
6   Neither disagree nor agree  Neither disagree nor agree   
7               Fully Disagree              Fully Disagree   
8                  Fully agree                       Agree   
9               Fully Disagree              Fully Disagree   
10              Fully Disagree                    Disagree   
11                 Fully agree                 Fully agree   
12              Fully Disagree              Fully Disagree   
13              Fully Disagree                    Disagree   
14  Neither disagree nor agree  Neither disagree nor agree   
15  Neither disagree nor agree  Neither disagree nor agree   
16                         NaN                         NaN   
17  Neither disagree nor agree  Neither disagree nor agree   
18                       Agree                       Agree   
19                    Disagree                    Disagree   
20                       Agree  Neither disagree nor agree   
21                    Disagree                    Disagree   
22              Fully Disagree  Neither disagree nor agree   
23  Neither disagree nor agree  Neither disagree nor agree   
24                    Disagree  Neither disagree nor agree   
25                    Disagree                    Disagree   
26                       Agree  Neither disagree nor agree   
27              Fully Disagree                    Disagree   
28              Fully Disagree  Neither disagree nor agree   
29  Neither disagree nor agree                 Fully agree   
30              Fully Disagree              Fully Disagree   
31                    Disagree              Fully Disagree   
32                       Agree                       Agree   
33              Fully Disagree                    Disagree   
34                 Fully agree                 Fully agree   
35                       Agree  Neither disagree nor agree   
36                    Disagree                       Agree   
37  Neither disagree nor agree  Neither disagree nor agree   
38                       Agree                       Agree   
39              Fully Disagree                    Disagree   
40                    Disagree  Neither disagree nor agree   
41                 Fully agree                 Fully agree   
42              Fully Disagree                    Disagree   
43              Fully Disagree              Fully Disagree   
44  Neither disagree nor agree  Neither disagree nor agree   
45  Neither disagree nor agree                       Agree   
46  Neither disagree nor agree  Neither disagree nor agree   
47              Fully Disagree                    Disagree   
48                       Agree                       Agree   
49  Neither disagree nor agree                 Fully agree   
50              Fully Disagree              Fully Disagree   
51              Fully Disagree              Fully Disagree   
52                    Disagree                    Disagree   
53  Neither disagree nor agree                       Agree   
54              Fully Disagree              Fully Disagree   

                 Teach_StimAsk                 Teach_Avail  
0               Fully Disagree              Fully Disagree  
1   Neither disagree nor agree  Neither disagree nor agree  
2                  Fully agree                 Fully agree  
3               Fully Disagree  Neither disagree nor agree  
4               Fully Disagree              Fully Disagree  
5                  Fully agree                 Fully agree  
6   Neither disagree nor agree  Neither disagree nor agree  
7               Fully Disagree              Fully Disag

In [3]:
selected_data['Teach_Motivate'].unique()

array(['Fully Disagree', 'Disagree', 'Fully agree',
       'Neither disagree nor agree', nan, 'Agree'], dtype=object)

In [5]:
coding = {'Fully Disagree': 0, 'Disagree': 1, 'Neither disagree nor agree': 2, 'Agree': 3, 'Fully agree': 4}

selected_data.replace(coding, inplace=True)
selected_data

<ipython-input-5-6b076452e207>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  selected_data.replace(coding, inplace=True)


Teach_Motivate  Teach_LinkTheory  Teach_StimAsk  Teach_Avail
0              0.0               0.0            0.0          0.0
1              1.0               1.0            2.0          2.0
2              4.0               4.0            4.0          4.0
3              0.0               2.0            0.0          2.0
4              0.0               0.0            0.0          0.0
5              4.0               4.0            4.0          4.0
6              2.0               2.0            2.0          2.0
7              0.0               0.0            0.0          0.0
8              4.0               3.0            3.0          4.0
9              0.0               0.0            1.0          3.0
10             0.0               1.0            0.0          2.0
11             4.0               4.0            4.0          4.0
12             0.0               0.0            0.0          3.0
13             0.0               1.0            NaN          3.0
14             2.0               2.0            2.0          2.0
15             2.0               2.0            2.0          2.0
16             NaN               NaN            NaN          NaN
17             2.0               2.0            3.0          3.0
18             3.0               3.0            2.0          3.0
19             1.0               1.0            1.0          2.0
20             3.0               2.0            2.0          2.0
21             1.0               1.0            3.0          2.0
22             0.0               2.0            0.0          3.0
23             2.0               2.0            2.0          4.0
24             1.0               2.0            1.0          2.0
25             1.0               1.0            0.0          3.0
26             3.0               2.0            4.0          4.0
27             0.0               1.0            1.0          1.0
28             0.0               2.0            0.0          0.0
29             2.0               4.0            2.0          4.0
30             0.0               0.0            0.0          0.0
31             1.0               0.0            0.0          0.0
32             3.0               3.0            3.0          3.0
33             0.0               1.0            0.0          2.0
34             4.0               4.0            4.0          4.0
35             3.0               2.0            3.0          4.0
36             1.0               3.0            2.0          3.0
37             2.0               2.0            2.0          2.0
38             3.0               3.0            3.0          3.0
39             0.0               1.0            3.0          4.0
40             1.0               2.0            0.0          0.0
41             4.0               4.0            2.0          4.0
42             0.0               1.0            1.0          1.0
43             0.0               0.0            0.0          1.0
44             2.0               2.0            3.0          3.0
45             2.0               3.0            2.0          2.0
46             2.0               2.0            1.0          2.0
47             0.0               1.0            2.0          4.0
48             3.0               3.0            3.0          3.0
49             2.0               4.0            4.0          4.0
50             0.0               0.0            0.0          2.0
51             0.0               0.0            0.0          3.0
52             1.0               1.0            2.0          2.0
53             2.0               3.0            3.0          3.0
54             0.0               0.0            0.0          0.0

In [7]:
selected_data.dropna(inplace=True)

<ipython-input-7-d426eaa94379>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  selected_data.dropna(inplace=True)


In [8]:
from scipy.stats import friedmanchisquare

friedmanchisquare(selected_data['Teach_Avail'], selected_data['Teach_Motivate'],
                  selected_data['Teach_StimAsk'], selected_data['Teach_LinkTheory'])


FriedmanchisquareResult(statistic=33.85869565217378, pvalue=2.1220432262951938e-07)

In [9]:
statistic=33.85869565217378
pvalue=2.1220432262951938e-07

In [10]:
# Effect size - Kendall W

n = selected_data.shape[0]
k = selected_data.shape[1]

W = statistic / (n * (k-1))
W

0.21294777139731935

In [11]:
W = 0.21294777139731935

In [12]:
if W < 0.3:
  print('Weak')
elif W < 0.5:
  print('Moderate')
elif W < 0.7:
  print('Relatively Strong')
else:
  print('Strong')

Weak


The difference were not having the same responses and after applying the Friedman test the difference was significant (p < 0.05) and further the Kendall W effect size suggests the difference is weak (W = 0.21)